lib/iron_bank/describe/excluded_fields.rb

```diff
@@ -87,9 +88,65 @@ def valid_query?
         info "Successful query for #{object_name}"
 
         true
-      rescue IronBank::InternalServerError, IronBank::BadRequestError => e
+      rescue IronBank::BadRequestError => e
         @last_failed_fields = extract_fields_from_exception(e)
 
+        false
+      rescue IronBank::InternalServerError
+        @last_failed_fields = exctract_from_dividing
```

Comment:
Extract `#extract_from_dividing` to service, so this line reads as:
```rb
@last_failed_fields = SearchFromQuery.call(object)
```
and the service class will have the form:
```rb
module IronBank
  module Describe
    class ExcludedFields::SearchFromQuery
      def self.call(object)
        new(object).call
      end
    
      def call
        # ... implementation here ...
      end

      private

      attr_reader :object

      def initialize(object)
        @object = object
      end
    end
  end
end
```

Reply:
with could also get away with:
```rb
rescue IronBank::BadRequestError, 
  IronBank::InternalServerError => error
  @last_failed_fields = ExtractFromMessage.call(error.message) ||
    DeduceFromQuery.call(object) 
end
```